app/src/main/java/com/hemdan/mvvm/ui/base/BaseViewModel.kt

```diff
@@ -17,7 +17,7 @@ import javax.inject.Inject
  * Email : mohammed.hemdan.faraj@gmail.com
  * Github : https://github.com/mhemdan
  */
-class BaseViewModel<Repository : BaseRepository> : ViewModel()  {
+open class BaseViewModel<Repository : BaseRepository> : ViewModel()  {
```

Comment:
@MonaAliMahmoud use abstract as it will be good to not create instance from BaseViewModel and for inheritance too so normally we use abstract for base classes